surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,361 @@
+# Survey questions
+One of the [compiler team's focuses for
+2022](https://blog.rust-lang.org/inside-rust/2022/02/22/compiler-team-ambitions-2022.html)
+is to improve the experience of debugging Rust code. We'd like to better
+understand your experiences using a debugger with Rust - what's been good,
+what's been bad and how your experience debugging Rust could be better.
+
+Unless you choose to enter your email, your answers will be anonymous. Any
+personal data you submit as a part of this survey will be handled in accordance
+with our policy:
+
+https://github.com/rust-lang/surveys/blob/main/documents/data-sharing-annual-community-survey.md
+
+We estimate it will take about ? minutes to complete.
+
+### Which operating systems do you use regularly for Rust debugging?
+Note: this is specifically about which systems you use for development and
+debugging not all the systems you target.
+
+Type: select all that apply (optional)
+
+- Linux
+- Windows
+- Windows Subsystem for Linux
+- Mac OS
+- Embedded devices
+- Other (open response)
+
+> *Justification:* Cohort analysis
+>
+> Debuggers are often platform-specific and various improvements that the
+> compiler team could make could only benefit a subset of all platforms, so
+> knowing what users are using is valuable.
+
+### How would you rate your Rust expertise?
+Type: select one
+
+- I can't read or write Rust
+- I can write simple programs in Rust
+- I can write useful, production-ready code but it is a struggle
+- I am productive writing Rust
+- I am an expert
+
+> *Justification:* Cohort analysis
+>
+> Users with different levels of experience are likely to face different
+> challenges.
+
+
+### What forms of debugging do you typically use with Rust?
+Type: select all that apply
+
+- `println!`/`log`/`tracing`/`dbg!`
+- Stack traces (printed by `panic!()`/`backtrace-rs`)
+- Traditional stepping debuggers (gdb/lldb/WinDbg/Visual Studio)
+- Record/replay debuggers (rr/WinDbg or Visual Studio with Time Travel Debugging)
+- High-level tracing (`dtrace`/`ebpf`/`ETW`)
+
+> *Justification:*
+>
+> Understand how users are debugging Rust code.
+
+### What uses of Rust are you debugging?
+Type: select all that apply (optional)
+
+- Single-threaded application
+- Multi-threaded application
+- Rust applications using `async`/`await`
```

Comment:
Should this include an application using futures but not async/await?

Reply:
Added note that this includes futures